### PR TITLE
[WWST-1676] Add fix for Watts update.

### PR DESCRIPTION
**Pull request by MGoralczykS**
@greens @MMateusiakS @MAblewiczS @PKacprowiczS 
Add fix for Watts update on GoControl Plug-in Switch.

### CODE REVIEW

**Review by tpmanley — CHANGES_REQUESTED** [review]
I'll reach out to you offline

File: devicetypes/smartthings/zwave-metering-switch.src/zwave-metering-switch.groovy [hunk 1]
```diff
@@ -28,8 +28,7 @@ metadata {
 		fingerprint inClusters: "0x25,0x32"
 		fingerprint mfr: "0086", prod: "0003", model: "0012", deviceJoinName: "Aeotec Micro Smart Switch"
 		fingerprint mfr: "021F", prod: "0003", model: "0087", deviceJoinName: "Dome On/Off Plug-in Switch"
-		fingerprint mfr: "0086", prod: "0103", model: "0060", deviceJoinName: "Aeotec Smart Switch 6"  //US
```
Comment (tpmanley) [inline devicetypes/smartthings/zwave-metering-switch.src/zwave-metering-switch.groovy]: Why are all these fingerprints removed?

File: devicetypes/smartthings/zwave-metering-switch.src/zwave-metering-switch.groovy [hunk 2]
```diff
@@ -223,10 +223,11 @@ def poll() {
...
 def refresh() {
 	log.debug "refresh()"
+	configure()
 	encapSequence([
-		zwave.switchBinaryV1.switchBinaryGet(),
-		meterGet(scale: 0),
-		meterGet(scale: 2)
+			zwave.switchBinaryV1.switchBinaryGet(),
+			meterGet(scale: 0)//,
+			//meterGet(scale: 2)
```
Comment (tpmanley) [inline devicetypes/smartthings/zwave-metering-switch.src/zwave-metering-switch.groovy]: Why was this removed?

File: devicetypes/smartthings/zwave-metering-switch.src/zwave-metering-switch.groovy [hunk 3]
```diff
@@ -223,10 +223,11 @@ def poll() {
 
 def refresh() {
 	log.debug "refresh()"
+	configure()
```
Comment (tpmanley) [inline devicetypes/smartthings/zwave-metering-switch.src/zwave-metering-switch.groovy]: Why was this added?

**Comment by MMateusiakS** [comment]
Hi, @tpmanley! Thanks for quick response! Michal is new member in our team, so he is in phase of learning git and developing DTH. So few bad things happened, but we have been fixing this issues meantime you have been reviewing.

**Comment by tpmanley** [comment]
@MMateusiakS totally understand! I'll let you finish up and will take another look when you're done.

File: devicetypes/smartthings/zwave-metering-switch.src/zwave-metering-switch.groovy [hunk 4]
```diff
@@ -118,11 +118,11 @@ def updated() {
 
 def getCommandClassVersions() {
 	[
-		0x20: 1,  // Basic
```
Comment (MMateusiakS) [inline devicetypes/smartthings/zwave-metering-switch.src/zwave-metering-switch.groovy:121]: This method is not used in your code. I think it is good practice for learning, but in final commit it is redundant.

File: devicetypes/smartthings/zwave-metering-switch.src/zwave-metering-switch.groovy [hunk 5]
```diff
@@ -145,13 +145,18 @@ def parse(String description) {
...
+			}
 			createEvent(name: "energy", value: cmd.scaledMeterValue, unit: "kWh")
 		} else if (cmd.scale == 1) {
 			createEvent(name: "energy", value: cmd.scaledMeterValue, unit: "kVAh")
 		} else if (cmd.scale == 2) {
-			createEvent(name: "power", value: Math.round(cmd.scaledMeterValue), unit: "W")
+			createEvent(name: "power", value: Math.round(cmd.scaledMeterValue), unit: "W", isStateChange: true)
 		}
 	}
+
```
Comment (MMateusiakS) [inline devicetypes/smartthings/zwave-metering-switch.src/zwave-metering-switch.groovy:159]: It is unnecessary empty line.

File: devicetypes/smartthings/zwave-metering-switch.src/zwave-metering-switch.groovy [hunk 6]
```diff
@@ -224,9 +229,8 @@ def poll() {
 def refresh() {
 	log.debug "refresh()"
 	encapSequence([
-		zwave.switchBinaryV1.switchBinaryGet(),
-		meterGet(scale: 0),
-		meterGet(scale: 2)
+			zwave.switchBinaryV1.switchBinaryGet(),
```
Comment (MMateusiakS) [inline devicetypes/smartthings/zwave-metering-switch.src/zwave-metering-switch.groovy:232]: why getMeter() with scale 2 was removed?

File: devicetypes/smartthings/zwave-metering-switch.src/zwave-metering-switch.groovy (hunk 4 above)
Comment (greens) [reply devicetypes/smartthings/zwave-metering-switch.src/zwave-metering-switch.groovy:121]: It's used in L134

File: devicetypes/smartthings/zwave-metering-switch.src/zwave-metering-switch.groovy [hunk 7]
```diff
@@ -145,13 +145,18 @@ def parse(String description) {
 def handleMeterReport(cmd){
 	if (cmd.meterType == 1) {
 		if (cmd.scale == 0) {
+			//automatically updates Watts on GoControl Plug-in Switch
+			if(zwaveInfo.mfr == "014F" && zwaveInfo.prod == "5053" && zwaveInfo.model == "3531") {
+				sendHubCommand(encap(meterGet(scale: 2)))
```
Comment (greens) [inline devicetypes/smartthings/zwave-metering-switch.src/zwave-metering-switch.groovy:150]: Are you doing this because the switch isn't reporting power, because this won't work with local execution. Your only option is to see if changing any configuration values prompts the device to report on its own:

https://products.z-wavealliance.org/products/1887/configs

Comment (MMateusiakS) [reply devicetypes/smartthings/zwave-metering-switch.src/zwave-metering-switch.groovy:150]: Michal and Mariusz were changing configuration, but it device did not response with Watts. As I know this is the way they figured out to get Watts update.

Comment (MGoralczykS) [reply devicetypes/smartthings/zwave-metering-switch.src/zwave-metering-switch.groovy:150]: I checked this with parameters. In configure() I added:
`else if (zwaveInfo.mfr == "014F" && zwaveInfo.prod == "5053" && zwaveInfo.model == "3531") {
    	result << response(encap(zwave.configurationV1.configurationSet(parameterNumber: 11, size: 1, scaledConfigurationValue: 1)))
        result << response(encap(zwave.configurationV1.configurationSet(parameterNumber: 12, size: 2, scaledConfigurationValue: 255)))
    }`

Watts didn't update. Also when I set `runLocally` to `true` ours solution didn't work. 
@greens maybe you have another idea for this?

Comment (greens) [reply devicetypes/smartthings/zwave-metering-switch.src/zwave-metering-switch.groovy:150]: @mckeed What can we do about this beyond just setting up polling?

Comment (mckeed) [reply devicetypes/smartthings/zwave-metering-switch.src/zwave-metering-switch.groovy:150]: @MGoralczykS have you tested while changing the power draw on the switch? In case does report power (watts) but only if it has changed significantly?

If it never reports power on its own then yet we'll have to poll like is done for GE devices.

Comment (MGoralczykS) [reply devicetypes/smartthings/zwave-metering-switch.src/zwave-metering-switch.groovy:150]: @mckeed yes. I tested this with connected devices and this plug didn't update watts. It only updates when I used refresh button (ST Classic)

**Comment by k-morawski** [comment]
https://github.com/SmartThingsCommunity/SmartThingsPublic/pull/3602
